packages/vite-node/src/source-map.ts

```diff
@@ -48,6 +48,11 @@ export function withInlineSourcemap(result: TransformResult, options: {
   while (OTHER_SOURCE_MAP_REGEXP.test(code))
     code = code.replace(OTHER_SOURCE_MAP_REGEXP, '')
 
+  // If the first line is not present on source maps, add simple 1:1 mapping ([0,0,0,0], [1,0,0,0])
+  // so that debuggers can be set to break on first line
+  if (map.mappings.startsWith(';'))
+    map.mappings = `AAAA,CAAA${map.mappings}`
```

Comment:
Don't we already modify source mapping for coverage support somewhere?

Reply:
In V8 provider, yes:

https://github.com/vitest-dev/vitest/blob/6eda473f30c4670ec72a06bd3b083a724712ae2a/packages/coverage-v8/src/provider.ts#L394-L406

Reply:
No, I mean don't we fill the map somewhere because of the vite-node offset? How is this PR different? Should we not fill it anymore?

Reply:
Vite-node's offset doesn't directly affect source maps in coverage. We just instruct `v8-to-istanbul` that the collected V8 coverage JSON reports include execution wrapper:

https://github.com/vitest-dev/vitest/blob/6eda473f30c4670ec72a06bd3b083a724712ae2a/packages/coverage-v8/src/provider.ts#L50-L51

https://github.com/vitest-dev/vitest/blob/6eda473f30c4670ec72a06bd3b083a724712ae2a/packages/coverage-v8/src/provider.ts#L362-L365

This PR has nothing to do with code coverage. It's about being able to set debugger statement on top of the file. You should be able to see this in VSCode's debugger as well.

This PR only affects `--inspect-brk` flag. It instructs debuggers to stop in the first test file before executing it. If the breakpoint is set on a line that is not included in source maps users will see transpiled code: https://github.com/vitest-dev/vitest/pull/5355#discussion_r1518540412. If the line is on source maps, users will see their original test file: https://github.com/vitest-dev/vitest/pull/5355#discussion_r1518541771. If Vite's SSR transformed import statements were included in source maps we could set breakpoint right after vite-node's wrapper - like line 3 or something. But as they are not, it's really difficult to try to guess how many import statements Vite added: https://github.com/vitest-dev/vitest/pull/5355#discussion_r1518560031. It's easiest to manually add breakpoint for source maps if there is none.